### PR TITLE
Add user to team/60

### LINKED ISSUE #60: Add user to team
What has to be done:
Add a player to a team

Why it has to be done:
So that the team captain can add players to his team with which they can participate to the team,

When the task is done:
The frontend application can call the backend application with all information to add a player to the team.

### LINKED ISSUE #82: GraphQL schema not showing
**Describe the bug**
During testing of the api, we found out that only one graphql schema is showing.
So it is only possible to get one type of data from the api (i.e. Teams)

### PR DESCRIPTION
**Pull request by lmtjanssen**
fixes #60 

also the API now runs and executes queries and mutations correctly and fixes #82

### CODE REVIEW

File: src/main/java/com/fontys/api/service/TeamService.java [hunk 1]
```diff
@@ -12,32 +13,42 @@
...
+
+    @Transactional
+    public String deleteTeam(Integer id) {
+        Optional<Team> team = teamRepository.findById(id);
+        team.ifPresent(teamRepository::delete);
+        return "Team " + team.get().getName() + " deleted";
+    }
+
+    @Transactional
+    public String addUserToTeam(Integer teamId, Integer userId) {
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/service/TeamService.java:51]: Manual testing shows that this function doesn't actually work

File: src/main/resources/graphql/schema.graphqls [hunk 2]
```diff
@@ -0,0 +1,22 @@
+type Query {
```
Comment (lfb0801) [inline src/main/resources/graphql/schema.graphqls:1]: position of the comments is changed when showing the schema, so it currently looks weird

File: src/main/resources/graphql/schema.graphqls [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    user(id:ID!):User
+    #team
+    teams(count: Int):[Team]
+    team(id:ID!):Team
+    #tournament
+    tournaments:[Tournament]
+    tournament(id:ID!):Tournament
+}
+
+type Mutation {
```
Comment (lfb0801) [inline src/main/resources/graphql/schema.graphqls:12]: position of the comments is changed when showing the schema, so it currently looks weird

File: src/main/java/com/fontys/api/entities/Team.java [hunk 4]
```diff
@@ -1,28 +1,36 @@
...
 import javax.persistence.GeneratedValue;
 import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.List;
 
 @Data
 @Entity
+@NoArgsConstructor
 public class Team {
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/entities/Team.java:15]: misses @AllArgsConstructor

File: src/main/java/com/fontys/api/mutations/TeamMutation.java [hunk 5]
```diff
@@ -19,7 +19,11 @@ public Team createTeam(String name) {
         return teamService.createTeam(name);
     }
 
-    public void deleteTeam(Long id) {
-        teamService.deleteTeam(id);
+    public String deleteTeam(Integer id) {
+        return teamService.deleteTeam(id);
+    }
+
+    public String addUserToTeam(Integer teamId, Integer userId) {
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/mutations/TeamMutation.java:26]: this function doesn't seem to do anything

**Review by lfb0801 — CHANGES_REQUESTED** [review]
implement requested changes and let's hope we never have to use mock after this. Also doesn't this pr fix #82 ?

File: src/main/java/com/fontys/api/service/TeamService.java (hunk 1 above)
Comment (lmtjanssen) [reply src/main/java/com/fontys/api/service/TeamService.java:51]: It only works if a user and team are created, as should always be the case

File: src/main/resources/graphql/schema.graphqls (hunk 2 above)
Comment (lmtjanssen) [reply src/main/resources/graphql/schema.graphqls:1]: Agreed, looking for a way to display it nicely

File: src/main/resources/graphql/schema.graphqls (hunk 3 above)
Comment (lmtjanssen) [reply src/main/resources/graphql/schema.graphqls:12]: Agreed, looking for a way to display it nicely

File: src/main/java/com/fontys/api/entities/Team.java (hunk 4 above)
Comment (lmtjanssen) [reply src/main/java/com/fontys/api/entities/Team.java:15]: I'll add it

File: src/main/java/com/fontys/api/mutations/TeamMutation.java (hunk 5 above)
Comment (lmtjanssen) [reply src/main/java/com/fontys/api/mutations/TeamMutation.java:26]: addUserToTeam or deleteTeam?

Comment (lfb0801) [reply src/main/java/com/fontys/api/mutations/TeamMutation.java:26]: addUserToTeam

File: src/main/java/com/fontys/api/service/TeamService.java (hunk 1 above)
Comment (lfb0801) [reply src/main/java/com/fontys/api/service/TeamService.java:51]: I tested the function in the right order, but if I did it wrong, then I should have gotten a usefull exception

**Comment by lmtjanssen** [comment]
Fixes #82 as well

**Review by lfb0801 — APPROVED** [review]

File: src/main/java/com/fontys/api/entities/Team.java [hunk 6]
```diff
@@ -1,28 +1,35 @@
 package com.fontys.api.entities;
 
+import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.Id;
+import javax.persistence.*;
```
Comment (larsvandriel) [inline src/main/java/com/fontys/api/entities/Team.java:7]: I think it's beter to have specific endpoint instead of importing the whole package at once

**Review by MichielLangen — APPROVED** [review]

**Pull request merged**